### PR TITLE
Add bridgeless support for dev-client on iOS

### PR DESCRIPTION
**Pull request by gabrieldonadel**
# Why

Closes ENG-11706

# How

- Reenable dev-client on fabric-tester
- Cleanup `DevClientAppDelegate` to use the same methods as `RCTAppDelegate`
- Refactor `DevMenuViewController.swift`

# Test Plan

Run BareExpo and FabricTester on iOS

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-dev-launcher/ios/EXDevLauncherController.m [hunk 1]
```diff
@@ -736,7 +736,7 @@ -(void)copyToClipboard:(NSString *)content {
 - (void)setDevMenuAppBridge
 {
   DevMenuManager *manager = [DevMenuManager shared];
-  manager.currentBridge = self.appBridge;
+  manager.currentBridge = [RCTBridge currentBridge];
```
Comment (Kudo) [inline packages/expo-dev-launcher/ios/EXDevLauncherController.m:739]: is that intentional? if `appBridge` is not being used, could we remove them?
also kinda afraid of the loading sequence between app bridge and dev launcher bridge. we cannot ensure RCTBridge.currentBridge is app bridge or EXDevLauncherRCTBridge.

**Review by Kudo — COMMENTED** [review]
generally looks okay. having a further question:
if we use RCTRootViewFactory now, do we still need DevClientAppDelegate or other app delegates? if we could directly have RCTRootViewFactory as a objective-c property and remove the AppDelegate usage, then we could reduce maintenance cost for react-native upgrade.

File: packages/expo-dev-launcher/ios/EXDevLauncherController.m (hunk 1 above)
Comment (gabrieldonadel) [reply packages/expo-dev-launcher/ios/EXDevLauncherController.m]: Reverted this back to `self.appBridge`

**Comment by gabrieldonadel** [comment]
> generally looks okay. having a further question: if we use RCTRootViewFactory now, do we still need DevClientAppDelegate or other app delegates? if we could directly have RCTRootViewFactory as a objective-c property and remove the AppDelegate usage, then we could reduce maintenance cost for react-native upgrade.

Yeah, I think we could potentially remove most of these other app delegates. I'll open a follow-up PR with doing that

**Review by Kudo — APPROVED** [review]

**Review by lukmccall — APPROVED** [review]

File: packages/expo-dev-launcher/ios/EXDevLauncherController.m [hunk 2]
```diff
@@ -598,7 +598,7 @@ - (void)_initAppWithUrl:(NSURL *)appUrl bundleUrl:(NSURL *)bundleUrl manifest:(E
 
     [self setDevMenuAppBridge];
 
-    [self _ensureUserInterfaceStyleIsInSyncWithTraitEnv:self.window.rootViewController];
+//    [self _ensureUserInterfaceStyleIsInSyncWithTraitEnv:self.window.rootViewController];
```
[the remark this answers is not in the document]
Comment (gabrieldonadel) [reply packages/expo-dev-launcher/ios/EXDevLauncherController.m:601]: This was causing a hard crash on new arch related to https://github.com/facebook/react-native/issues/42120. From my testing, we can just remove this method

https://github.com/expo/expo/assets/11707729/34bd725c-3199-4e28-b27a-ae50be1b96f3

**Pull request merged**